test/rules/strict-comparisons/default/test.ts.lint

```diff
@@ -188,3 +188,17 @@ const func = (param: string | null = null) => {
 
 [CANNOT_USE]: Cannot use '%s' comparator for type '%s'.
 [CANNOT_COMPARE]: Cannot compare type '%s' to type '%s'.
+
+const MyType = string | boolean | number
+
+class MyClass<T extends MyType> {
```

Comment:
I suspect there are edge cases we'll want to cover here. Would you mind adding the following?

* Generic `function () {}`s and `() => {}`s
* Closures with multiple generic types: i.e. `<T, U>`
* Generics with constraints that _do_ match up: i.e. `<T extends number, U extends number>`
* Generics with constraints that _don't_ match up: i.e. `<T extends number, U extends string>`